src/main/java/picard/sam/markduplicates/UmiUtil.java

```diff
@@ -34,10 +36,55 @@
 
 public class UmiUtil {
 
-    /** Returns an instance of the UMI without dashes */
-    public static String getSanitizedUMI(final SAMRecord record, final String umiTag) {
-        String umi = record.getStringAttribute(umiTag);
+    /**
+     * Creates a top-strand normalized duplex UMI.
+     * Duplex UMIs that come from a top strand read are by definition, top-strand normalized.
+     * The UMI from a bottom strand can be normalized to be identical to the UMI
+     * read from its corresponding top strand by swapping the content of the
+     * UMI around the "-" found in duplex UMIs.  For example, a bottom strand
+     * UMI reading ATC-CGG when top-strand normalized will read CGG-ATC.
+     *
+     * @param record SAM record to retrieve UMI from.
+     * @param umiTag The tag used in the bam file that designates the UMI.
+     * @return Normalized Duplex UMI.  If the UMI isn't duplex, it returns the UMI unaltered.
+     */
+    static String getTopStrandNormalizedDuplexUMI(final SAMRecord record, final String umiTag) {
+        final String umi = record.getStringAttribute(umiTag);
+
         if (umi == null) return null;
-        return umi.replace("-", "");
+
+        final String[] split = umi.split("-");
```

Comment:
Same issue here with magic strings, is there a constant for `"-"` to make all common uses of it clearer?